### PR TITLE
Closes #73 - Created `adsl_metabolic`

### LINKED ISSUE #73: Feedback from testing:  Provide an `adsl_metabolic` rather than a `dm_metabolic`.
### Background Information

If we provide an `adsl_metabolic` dataset instead of `dm_metabolic`, then the code in the "Read in Data" section of the vignettes could be simplified.

### Definition of Done

- [ ] Evaluate if we want to make this change
- [ ] Make change or close issue

### PR DESCRIPTION
**Pull request by AndersAskeland**
Closes #73 Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral<ext> (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by AndersAskeland — COMMENTED** [review]
I dont have R available, so I would like someone else to also take a look. But I think everything here is based on `admiral_adsl`, where the code is basically copied, and you have done a filter in the end?

I have one thing that needs to be changed, which is you need to run `styler::style_file()` on the file, and commit/push the changes generated. That should fix the style error.

File: inst/templates/ad_adsl_metabolic.R [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+ex <- pharmaversesdtm::ex
+ae <- pharmaversesdtm::ae
+lb <- pharmaversesdtm::lb
+
+# When SAS datasets are imported into R using haven::read_sas(), missing
+# character values from SAS appear as "" characters in R, instead of appearing
+# as NA values. Further details can be obtained via the following link:
+# https://pharmaverse.github.io/admiral/articles/admiral.html#handling-of-missing-values # nolint
+
+
```
Comment (yurovska) [inline inst/templates/ad_adsl_metabolic.R:29]:
```suggestion
```

File: inst/templates/ad_adsl_metabolic.R [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+                     str_detect(EXTRT, "PLACEBO"))) & !is.na(EXENDTM),
+  new_vars = exprs(TRTEDTM = EXENDTM, TRTETMF = EXENTMF),
+  order = exprs(EXENDTM, EXSEQ),
+  mode = "last",
+  by_vars = exprs(STUDYID, USUBJID)
+) %>%
+  ## Derive treatment end/start date TRTSDT/TRTEDT ----
+derive_vars_dtm_to_dt(source_vars = exprs(TRTSDTM, TRTEDTM)) %>%
+  ## derive treatment duration (TRTDURD) ----
+derive_var_trtdurd()
```
Comment (yurovska) [inline inst/templates/ad_adsl_metabolic.R:153]:
```suggestion
adsl <- dm_metabolic %>%
  ## derive treatment variables (TRT01P, TRT01A) ----
  # See also the "Visit and Period Variables" vignette
  # (https://pharmaverse.github.io/admiral/articles/visits_periods.html#treatment_adsl)
  mutate(TRT01P = ARM, TRT01A = ACTARM) %>%
  ## derive treatment start date (TRTSDTM) ----
  derive_vars_merged(
    dataset_add = ex_ext,
    filter_add = (EXDOSE > 0 |
      (EXDOSE == 0 &
        str_detect(EXTRT, "PLACEBO"))) &
      !is.na(EXSTDTM),
    new_vars = exprs(TRTSDTM = EXSTDTM, TRTSTMF = EXSTTMF),
    order = exprs(EXSTDTM, EXSEQ),
    mode = "first",
    by_vars = get_admiral_option("subject_keys")
  ) %>%
  ## derive treatment end date (TRTEDTM) ----
  derive_vars_merged(
    dataset_add = ex_ext,
    filter_add = !is.na(EXSTDTM) & (
      EXDOSE > 0 | (EXDOSE == 0 & str_detect(EXTRT, "PLACEBO"))
    ),
    new_vars = exprs(TRTEDTM = EXENDTM, TRTETMF = EXENTMF),
    order = exprs(EXENDTM, EXSEQ),
    mode = "last",
    by_vars = get_admiral_option("subject_keys")
  ) %>%
  ## Derive treatment end/start date TRTSDT/TRTEDT ----
  derive_vars_dtm_to_dt(source_vars = exprs(TRTSDTM, TRTEDTM)) %>%
  ## derive treatment duration (TRTDURD) ----
  derive_var_trtdurd()
```

**Review by yurovska — CHANGES_REQUESTED** [review]
I remember we discussed during one of our meetings that we wanted to add a sort of dummy diabetic status in `adsl_metabolic` required for some derivations proposed by @starosto. Is this still relevant?

File: inst/templates/ad_adsl_metabolic.R (hunk 2 above)
Comment (yurovska) [reply inst/templates/ad_adsl_metabolic.R:153]: @AndersAskeland  This should resolve the styler issue

**Comment by manciniedoardo** [comment]
Update from today's call: @starosto to use temporary copy of adsl_metabolic within ADLB vignette branch in order for this PR not to be a blocker for the ADLB vignette.

**Comment by AndersAskeland** [comment]
> Hi all, I have solved the style error, but not roxygen error. Should manuals be updated? man/derive_param_ratio.Rd man/derive_param_waisthgt.Rd man/derive_param_waisthip.Rd

Yeah, absolutely correct. I just went ahead and updated it and will merge it 👍  Thanks so much for your work on this
